dashbuilder-shared/dashbuilder-dataset-shared/src/main/java/org/dashbuilder/dataset/engine/function/MaxFunction.java

```diff
@@ -34,34 +33,36 @@ public AggregateFunctionType getType() {
         return AggregateFunctionType.MAX;
     }
 
-    public double aggregate(List values) {
-        if (values == null || values.isEmpty()) return 0;
-
-        // Get the max. value from the collection.
-        Number max = null;
-        Iterator it = values.iterator();
-        while (it.hasNext()) {
-            Number n = (Number) it.next();
-            if (n == null) continue;
-            if (max == null || n.doubleValue() > max.doubleValue()) max = n;
+    public Object aggregate(List values) {
+        if (values == null || values.isEmpty()) {
+            return null;
         }
-        if (max == null) return 0;
-        return round(max.doubleValue(), precission);
+
+        List sorted = (List) values.stream().sorted().collect(Collectors.toList());
```

Comment:
As mentioned above: Please filter out nulls and provide some unit tests to cover those cases.